contributors/devel/pull-requests.md

```diff
@@ -1,111 +1,326 @@
+  * [Automation](#automation)
+  * [How the e2e Tests Work](#how-the-e2e-tests-work)
+- [Why was my PR closed?](#why-was-my-pr-closed-)
+- [Why is my PR not getting reviewed?](#why-is-my-pr-not-getting-reviewed-)
+- [Best Practices for Faster Reviews](#best-practices-for-faster-reviews)
+  * [0. Familiarize yourself with project conventions](#0-familiarize-yourself-with-project-conventions)
+  * [1. Is the feature wanted? Make a Design Doc or Sketch PR](#1-is-the-feature-wanted--make-a-design-doc-or-sketch-pr)
+  * [2. Smaller Is Better: Small Commits, Small PRs](#2-smaller-is-better--small-commits--small-prs)
+  * [3. Open a Different PR for Fixes and Generic Features](#3-open-a-different-pr-for-fixes-and-generic-features)
+  * [4. Comments Matter](#4-comments-matter)
+  * [5. Test](#5-test)
+  * [6. Squashing and Commit Titles](#6-squashing-and-commit-titles)
+  * [7. KISS, YAGNI, MVP, etc.](#7-kiss--yagni--mvp--etc)
+  * [8. It's OK to Push Back](#8-it-s-ok-to-push-back)
+  * [Common Sense and Courtesy](#common-sense-and-courtesy)
+<!-- END MUNGE: GENERATED_TOC -->
 
-- [Pull Request Process](#pull-request-process)
-- [Life of a Pull Request](#life-of-a-pull-request)
-  - [Before sending a pull request](#before-sending-a-pull-request)
-  - [Release Notes](#release-notes)
-    - [Reviewing pre-release notes](#reviewing-pre-release-notes)
-  - [Visual overview](#visual-overview)
-- [Other notes](#other-notes)
-- [Automation](#automation)
+# Pull Request Process - Who is this for?
 
-<!-- END MUNGE: GENERATED_TOC -->
+This doc explains the process and best practices for submitting a PR to the [Kubernetes project](https://github.com/kubernetes/kubernetes). It should serve as a reference for all contributors, and be useful especially to new and infrequent submitters.
+
+# Before You Submit a PR - Prerequisites
+
+If you're just getting started, please refer to the [development guide](development.md) to set up your local dev environment and get coding. This guide is for contributors who already have a PR to submit.
+
+**Make sure your PR adheres to our best practices. These include following project conventions, making small PRs, and commenting thoroughly. Please read the more detailed section on [Best Practices for Faster Reviews](#best-practices-for-faster-reviews) at the end of this doc.**
+
+## Run Local Verifications
+
+Run these local verifications before you submit your PR.
+
+* Enable, run, and pass the Kubernetes [pre-commit hook](development.md#define-a-pre-commit-hook) which checks your repo formatting (and more); note that you may want to add these when you're closer to the end of your project, as they take time to run every time you commit
+* Run and pass `make verify` (can take 30-40 minutes)
+* Run and pass `make test`
+* Run and pass `make test-integration`
+
+## Sign the CLA
+
+You must sign the CLA before your first contribution. [Read more about the CLA.](https://github.com/kubernetes/community/blob/master/CLA.md)
```

Comment:
Spell out "Content License Agreement (CLA)" the first time you use it in a sentence.